docs/contributing/test-suite.rst

```diff
@@ -7,10 +7,36 @@
 **********
 Test suite
 **********
+To run the the test suite you'll need nox__, an automated testing tool for Python.
```

Comment:
`nox` is not required to run the tests. I don't use it and I would not even notice if the `nox` configuration in the repository would break. I don't feel recommending it as the tool of choice to run the tests. Plain `pytest` is probably a more conservative choice.

Also, I don't know `nox`, but from the configuration file I deduce that it requires installing `meson-python` for every test invocation. This is not necessary and unnecessarily slow.

Reply:
Shall I keep the nox content (i guess useful for testing against a number of python versions..?) and prefix it with standard pytest?

Reply:
That sounds good to me. I'll note that the "set up your development environment" part, which is a prerequisite for a basic:
```
$ pip install -e . --no-build-isolation
$ pytest
```
is also still empty. I suggest we discuss that in a separate issue though, because it may involve either longer install instructions or duplicating dependencies, and there's a bunch of ways of doing this.

Reply:
If that sounds good, I can open a new issue with a proposal.

Reply:
> That sounds good to me. I'll note that the "set up your development environment" part, which is a prerequisite for a basic:

There is no need to install `meson-python` to run the tests. I never do. Simply running `pytest` in the project source directory works for me.

Reply:
Oh, sure. Sorry, I think I misinterpreted your comment. But you would get a very similar error if you try to run `pip install -e .` with the `--no-build-isolation` option.

Reply:
Yes, agreed. It's a separate topic, but a prerequisite for running tests at all. That's why I suggested keeping it to a separate issue/PR. "setting up a dev env" tends to be a hotly debated topic:)

Reply:
> "setting up a dev env" tends to be a hotly debated topic

That's why I'm not sure about covering it in our documentation. I already been deemed not worthy to work on `meson-python` because I don't use `pre-commit-hook`. I don't want to have to read similar things because the way to setup a development environment we suggest in the documentation does not reflect what is trendy this week. 😆

Reply:
Yeah, I definitely don't want it to be prescriptive. As a conda/mamba user I'm often left out in the cold too by other projects. I'd more like a few helpful hints like "here is what you need, you can set it up like _this_, or _that_, or any way you like as long as you have all the needed dev dependencies".

Reply:
"The testsuite uses pytest. The easiest way to run pytest is via the provided `nox` testing tool, which handles all the nitty gritty details for you including installation of dev dependencies and much more. [rest of description]"

Seems reasonable?